src/EditorFeatures/CSharp/MisplacedUsings/MisplacedUsingsCodeFixProvider.SourceMap.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.MisplacedUsings
+{
+    /// <summary>
+    /// Implements a code fix for all misplaced using statements.
+    /// </summary>
+    internal partial class MisplacedUsingsCodeFixProvider
+    {
+        /// <summary>
+        /// Contains a map of the different regions of a source file.
+        /// </summary>
+        /// <remarks>
+        /// <para>Used source file regions are:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description>conditional directives (#if, #else, #elif, #endif)</description></item>
+        /// <item><description>pragma warning directives</description></item>
+        /// <item><description>region directives</description></item>
+        /// </list>
+        /// </remarks>
+        private class SourceMap
```

Comment:
this info feels like it could be very redundant with the info we compute in DirectiveSyntaxExtensions.